src/smtp.rs

```diff
@@ -28,11 +27,11 @@ pub enum Error {
         #[source]
         error: error::Error,
     },
-    #[error("SMTP: failed to connect: {0}")]
+    #[error("SMTP failed to connect: {0}")]
     ConnectionFailure(#[source] smtp::error::Error),
-    #[error("SMTP: failed to setup connection {0:?}")]
+    #[error("SMTP failed to setup connection: {0}")]
     ConnectionSetupFailure(#[source] smtp::error::Error),
-    #[error("SMTP: oauth2 error {address}")]
+    #[error("SMTP oauth2 error {address}")]
```

Comment:
We can discuss about this, but I think it makes the error (that's now shown to the user) more readable. It's shorter and colons are used to separate error and error source.

Before:
SMTP: failed to setup connection Io(Custom { kind: Other, error: VerboseError { source: Custom { kind: Other, error: "failed to lookup address information: No address associated with hostname" }, message: "could not resolve address `("testrun.org", 587)`" } }))

After:
SMTP failed to setup connection: io: could not resolve address `("testrun.org", 587)`